### PR TITLE
Registry code cleanup for Fake backend

### PR DESCRIPTION
**Pull request by diegomarangoni**
EES-4629

This PR removes all unnecessary code after the complete migration to registry pattern related to Fake backend.

### CODE REVIEW

**Comment by vgarvardt** [comment]
👍

**Review by boekkooi-fresh — COMMENTED** [review]
Why did you not just remove `pkg/backends/fake/fake.go`?

File: pkg/backends/fake/backend_test.go [hunk 1]
```diff
@@ -38,8 +38,11 @@ func TestCheckOrCreateResources(t *testing.T) {
 			return true, nil, &StatusError{}
 		})
 
-		backend := New(client, lt, zap.NewNop())
-		assert.NoError(t, backend.CheckOrCreateResources(context.TODO()))
+		backend := &Backend{
+			kubeClient: client,
+			logger:     zap.NewNop(),
```
Comment (vgarvardt) [inline pkg/backends/fake/backend_test.go:43]:
```suggestion
			logger:     zaptest.NewLogger(t),
```

File: pkg/backends/fake/backend_test.go [hunk 2]
```diff
@@ -238,7 +260,28 @@ func TestCheckOrUpdateStatus(t *testing.T) {
...
+		Type:            "Fake",
+		Overwrite:       true,
+		DistributedPods: &distributedPods,
+		Tags:            loadTestV1.LoadTestTags{"team": "kangal"},
+	}
+
+	b := Backend{}
+	err := b.TransformLoadTestSpec(&spec)
+
+	assert.NoError(t, err)
```
Comment (vgarvardt) [inline pkg/backends/fake/backend_test.go:284]: Does it make sense to run other asserts if this one failed? If not then
```suggestion
	require.NoError(t, err)
```

Comment (diegomarangoni) [reply pkg/backends/fake/backend_test.go]: Nope, I changed

**Comment by vgarvardt** [comment]
👍

**Pull request merged**